src/api/aneel/datastore/search.py

```
@@ -0,0 +1,212 @@
+            func (callable): A função a ser decorada com a lógica de retentativa.
+
+        Returns:
+            callable: A função 'wrapper' que encapsula a lógica de retentativa.
+        """
+        def wrapper(*args, **kwargs):
+            retry_count = 0
+            while True:
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    _warnings.warn(f"Tentativa {retry_count+1}/{Search.RETRY}: {e}")
```

Ruff (0.12.2)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)